main.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# standard imports
+from datetime import datetime as dt
+
+# third party imports
+import click
+
+# first party imports
+from utilities import DataCleaner
+from utilities import QueryManager
+from utilities import Query
+from utilities import QuerySplitter
+from utilities.semaphore import ThreadManager
+
+
+@click.command()
+@click.option("-a", "--api-endpoint", default=None, required=True, help="api endpoint to query against")
+@click.option(
+    "-c",
+    "--cert",
+    default=None,
+    help="relative path to the certificate which is used to create the request",
+)
+@click.option(
+    "-t",
+    "--timeout",
+    default=30,
+    help="number of seconds the client will wait for the server to send a response",
+    show_default=True,
+    type=int,
+)
+@click.option(
+    "-k",
+    "--kwargs",
+    default=None,
+    help="parameters for the query; supported keys: target, params\ntarget > specifies a target behind the api endpoint\nparams > sets specific parameters for the query\n\tsupported parameters are:\n\t - 'query'\n\t - 'dedup'\n\t - 'partial_response'\n\t - 'step'\n\t - 'max_source_resolution'\n\t - 'engine'\n\t - 'analyze'",
+)
+@click.option("-p", "--directory-path", default=None, help="directory path in which the query results are stored")
+@click.option(
+    "-b",
+    "--threshold",
+    default=None,
+    help="Threshold in days which specifies when the data are interpolated by Thanos\nThis helps splitting the queries due to efficiency and resource optimization",
+    type=int,
+)
+@click.option(
+    "-d",
+    "--delay",
+    default=0.25,
+    help="Delay in seconds between each query execution",
+    type=int,
+)
+@click.option(
+    "-x",
+    "--threads",
+    default=12,
+    help="Maximum number of threads to use for query execution",
+    show_default=True,
+    type=int,
+)
+# TODO: add possible option for max long term storage, currently fixed at 5y
+def main(
+    api_endpoint: str = None,
+    cert: str = None,
+    timeout: int = 30,
+    kwargs: dict = None,
+    directory_path: str = None,
+    threshold: int = None,
+    delay: int = 0.25,
+    threads: int = 12,
+):
```

Comment:
You're setting the defaults in click already.

```suggestion
    api_endpoint: str = None,
    cert: str = None,
    timeout: int = None,
    kwargs: dict = None,
    directory_path: str = None,
    threshold: int = None,
    delay: float = None,
    threads: int = None,
):
```